h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -217,6 +208,24 @@ else if (c instanceof C8DChunk)
     return this;
   }
 
+  private long processUUIDchunk(Chunk c, long checksum, long start, long l) {
+    for( int i=c.nextNZ(-1); i< c._len; i=c.nextNZ(i) ) {
+      if( c.isNA(i) ) _naCnt++;
+      else {
+        try {
+          long lo = c.at16l(i), hi = c.at16h(i);
+          if (lo != 0 || hi != 0) _nzCnt++;
+          l = lo ^ 37*hi;
+        } catch (Chunk.WrongType wt) {
```

Comment:
I think the issue is in FVecParseWriter. The ID column of the dataset is not UUID but it actually contains numbers. CsvParser calls addNumCall and numerical values are added to the Chunk.

After the parse is done the Vec is declared as UUID but it contains numerical Chunks.

In my opinion this should be fixed in FVecParseWriter